profile/src/main/java/com/scottlogic/deg/profile/reader/file/names/NameRetriever.java

```diff
@@ -33,19 +35,28 @@ private NameRetriever() {
         throw new UnsupportedOperationException("No static class instantiation");
     }
 
-    public static Set<Object> loadNamesFromFile(NameConstraintTypes configuration) {
+    public static DistributedSet<Object> loadNamesFromFile(NameConstraintTypes configuration) {
         if (configuration == FULL) {
-            return new HashSet<>(combineFirstWithLastNames(
+            return downcastToObject(combineFirstWithLastNames(
                 generateNamesFromSingleFile(FIRST.getFilePath()),
                 generateNamesFromSingleFile(LAST.getFilePath())));
         } else {
-            return new HashSet<>(generateNamesFromSingleFile(configuration.getFilePath()));
+            return downcastToObject(generateNamesFromSingleFile(configuration.getFilePath()));
         }
     }
 
-    private static Set<String> generateNamesFromSingleFile(String source) {
+    private static <T> DistributedSet<Object> downcastToObject(DistributedSet<T> higher) {
```

Comment:
This is necessary because we are casting a reified type. The better solution here would be to use concrete types throughout the application instead of `Object`.